packages/medusa/src/services/event-bus.ts

```diff
@@ -151,22 +151,32 @@ export default class EventBusService
           },
         ]
 
-    /**
-     * We store events in the database when in an ongoing transaction.
-     *
-     * If we are in a long-running transaction, the ACID properties of a
-     * transaction ensure, that events are kept invisible to the enqueuer
-     * until the transaction has committed.
-     *
-     * This patterns also gives us at-least-once delivery of events, as events
-     * are only removed from the database, if they are successfully delivered.
-     *
-     * In case of a failing transaction, jobs stored in the database are removed
-     * as part of the rollback.
-     */
-    const stagedJobs = await this.stagedJobService_
-      .withTransaction(manager)
-      .create(events)
+    events = events.filter(
+      (event) =>
+        this.eventBusModuleService_?.retrieveSubscribers(event.eventName)
```

Comment:
suggestion(non-blocking): make `retrieveSubscribers` a bulk-fetch method optionally to get all event handlers at once rather than multiple round-trips since it's a module